keps/sig-network/1880-service-subnet/README.md

```diff
@@ -0,0 +1,558 @@
+
+```go
+// IPToDecimal converts an IP to a string with its decimal representation
+func IPToDecimal(ip net.IP) string {
+	if ip == nil {
+		return ""
+	}
+	i := new(big.Int)
+	// compare the address with its decimal representation
+	if ip.To4() != nil {
+		i.SetBytes(ip.To4())
+	} else if ip.To16() != nil {
+		i.SetBytes(ip.To16())
+	}
+	return i.String()
+}
+
+// DecimalToIP converts a string with the decimal representation of an IP address and returns the IP
+func DecimalToIP(ipDecimal string) net.IP {
+	if len(ipDecimal) == 0 {
+		return nil
+	}
+	// convert decimal representation to BigInt
+	i, ok := new(big.Int).SetString(ipDecimal, 10)
+	// if we can not convert the string to a number
+	// or the number has more than 16 bytes it is not an IP
+	if !ok || len(i.Bytes()) > 16 {
+		return nil
+	}
+	bufLen := net.IPv4len
+	if len(i.Bytes()) > net.IPv4len {
+		bufLen = net.IPv6len
+	}
+	// convert BigInt to IP address, truncate to IP Len bytes
+	r := append(make([]byte, bufLen), i.Bytes()...)
+	return net.IP(r[len(r)-bufLen:])
+}
+```
+
+
+```
+<<[UNRESOLVED IPAddress API Operations]>>
+
+* IPAddress objects created must reference the IPRange object, and validated against it. 
+The `address` field will contain the IP address representation, if no one is provided it will be defaulted useing the value obtained from the name. If the `address` is set it must match the `name`.
+
+* IPAddress status updates depend on the allocation model chosen, it may allowed to change after creation,
+
+QUESTION garbage collection, cluster and namespaced scoped objects. Which will be the best scope for these new objects? If we can make that IPAddresses are owned by the Services that allocate them, the garbage collector when deleting a Service will take care of the release????
+
+<<[/UNRESOLVED]>>
+```
+
+### NodePort
+
+TBD
+
+
+### Test Plan
+
+* Unit tests
+
+TBD
+
+* Integration Tests
+
+TBD
+
+* E2E tests 
+
+
+### Graduation Criteria
+
+The change must be transparent to the users, so there should not be any performance degradation
+or Services breaking changes. 
+
+### Upgrade / Downgrade Strategy
+
+The service-subnet and node-port range will keep using the flags for the initial configuration, it can
+happen that the service subnet or node-port range are modified after the initial configuration, in that
+case they will use the same value as configured by the flags.
+
+### Version Skew Strategy
+
+That apisever control-plane will keep using repair loops to sync current service IP allocated with
+the corresponding allocator, that guarantee that the skeweved apiservers work because they have the
+same source of truth.
+
+```
+<<[UNRESOLVED detect apiservers with different version ]>>
+
+Any update on the IPRange allocated object will not be allowed if there are apiservers without the
+same version
+QUESTION Is this even possible?
```

Comment:
this needs to be worked throug, for sure.  Not sure it matters, as long as the conversion uses the same values...